File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> left_half(m);
        vector<int> right_half(n);

        for (int i = 0; i < m; ++i)
            left_half[i] = nums1[i];
        for (int j = 0; j < n; ++j)
            right_half[j] = nums2[j];
        
        int i = 0;
        int j = 0;
        int k = 0;

        while (i < m && j < n) {
            if (left_half[i] <= right_half[j]) {
                nums1[k] = left_half[i];
                ++i;
            } else {
                nums1[k] = right_half[j];
                ++j;
            }
            ++k;
        }

        while (i < m) {
            nums1[k] = left_half[i];
            ++i;
            ++k;
        }

        while (j < n) {
            nums1[k] = right_half[j];
            ++j;
            ++k;
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.cs
public class Solution {
    public void Merge(int[] nums1, int m, int[] nums2, int n) {
        List<int> leftHalf = new List<int>(m);
        List<int> rightHalf = new List<int>(n);

        for (int i = 0; i < m; ++i)
            leftHalf.Add(nums1[i]);
        for (int j = 0; j < n; ++j)
            rightHalf.Add(nums2[j]);

        int leftIndex = 0;
        int rightIndex = 0;
        int k = 0;

        while (leftIndex < m && rightIndex < n)
        {
            if (leftHalf[leftIndex] <= rightHalf[rightIndex])
            {
                nums1[k] = leftHalf[leftIndex];
                ++leftIndex;
            }
            else
            {
                nums1[k] = rightHalf[rightIndex];
                ++rightIndex;
            }
            ++k;
        }

        while (leftIndex < m)
        {
            nums1[k] = leftHalf[leftIndex];
            ++leftIndex;
            ++k;
        }

        while (rightIndex < n)
        {
            nums1[k] = rightHalf[rightIndex];
            ++rightIndex;
            ++k;
        }
    }
}